checkups/kubevirt-vm-latency/vmlatency/internal/checkup/checkup.go

```
@@ -62,6 +62,14 @@ func New(c vmi.KubevirtVmisClient, namespace string, params config.CheckupParame
 }
 
 func (c *checkup) Preflight() error {
+	var err error
+	if _, err = c.client.GetNetworkAttachmentDefinition(
+		c.params.NetworkAttachmentDefinitionNamespace,
+		c.params.NetworkAttachmentDefinitionName,
+	); err != nil {
```

I think the preflight check should be specific about the missing entity, not if there are communication or access problems.
I think you should fail only in one condition, anything else should be ignored.

---

If there are communication / access issues at such early stage why continue the flow?
It is likely that there will be more issues later on that will cause the checkup to fail.
Failing faster will improve the UX, its one of the reasons for having preflight in the first place.

---

The short answer is: This is not the k8s way.
The long answer is: Accessing the NAD directly is not a must to pass the checkup at this moment. Furthermore, it is not the preflight interest about intermediate disconnection or unavailability of the API server.
When the parsing of the NAD is introduced, it makes sense to fail there if the data is not collected, not in a different place.

---

Initially this commit was part of #94, I was asked to move it to another PR, now it lake context and we have this discussion that could be obsolete.
With #94 PR changes fetching the NAD is a must.

Think about this scenario, the NAD is not exist and there was an issue with accessing the cluster API server,
the preflight will pass (false positive) and the checkup setup will later on fail due to missing data / NAD not found error.
Its wrong, isnt it?

---

And at the same time, it is not that accurate to fail due to a glitch on the API server.
There are a wide range of options to choose from, including retry, fail only on things you care about (I guess that is my suggestion) or just fail on first attempt as you suggested.

I do not know how a different action in a different PR will influence this decision.
But it does influence what fails and when.

I guess there is no one best path here, all have costs.
I am fine keeping it as is, thanks.